### ISSUE TITLE
AllowPreRelease switch does not work in PowerShell 6.1.1 on Ubuntu 18.04 in WSL

### ISSUE THREAD

**Issue by bergmeister** [PowerShell/PowerShellGetv2#388]
Steps to reproduce
------------------

```powershell
 install-Module pester -Verbose -AllowPrerelease -scope currentuser -whatif
```

Note that even when not using the `-WhatIf` switch, it cannot find the new pre-release.

Expected behavior
-----------------

```none
What if: Performing the operation "Install-Module" on target "Version '4.5.0-beta1' of module 'Pester'".
```

Actual behavior
---------------

```none
VERBOSE: Using the provider 'PowerShellGet' for searching packages.
VERBOSE: The -Repository parameter was not specified.  PowerShellGet will use all of the registered repositories.
VERBOSE: Getting the provider object for the PackageManagement Provider 'NuGet'.
VERBOSE: The specified Location is 'https://www.powershellgallery.com/api/v2' and PackageManagementProvider is 'NuGet'.
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='pester'' for ''.
VERBOSE: Total package yield:'1' for the specified package 'pester'.
VERBOSE: Skipping installed module Pester 4.4.3.
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.1
PSEdition                      Core
GitCommitId                    6.1.1
OS                             Linux 4.4.0-17763-Microsoft #194-Microsoft Mon Dec 03 17:58:00 PST 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

```powershell
> Get-Module -ListAvailable PowerShellGet,PackageManagement


    Directory: /home/chris/.local/share/powershell/Modules


ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Script     1.2.4      PackageManagement                   Desk      {Find-Package, Get-Package, Get-PackageProvider,...
Script     2.0.3      PowerShellGet                       Desk      {Find-Command, Find-DSCResource, Find-Module, Fi...


    Directory: /opt/microsoft/powershell/6/Modules


ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Script     1.1.7.2    PackageManagement                   Desk      {Find-Package, Get-Package, Get-PackageProvider,...
Script     1.6.7      PowerShellGet                       Desk      {Find-Command, Find-DSCResource, Find-Module, Fi...
```

```powershell
> Get-PackageProvider -ListAvailable
Name                     Version          DynamicOptions
----                     -------          --------------
NuGet                    3.0.0.1          Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag...
PowerShellGet            2.0.3.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, ...
PowerShellGet            1.6.7.0
```

**Comment by edyoung** [comment]
confirmed. The weird thing is, PowerShellGet has no idea about WSL. Something in a lower layer must be behaving differently between Ubuntu on WSL and Ubuntu on Linux, but I'm not sure what. See also #353 which also misbehaves mysteriously on WSL only.

**Comment by edyoung** [comment]
Finding this hard to repro now. Pester have published new versions so the newest build is no longer a prerelease. I tried another module which has a prerelease as its latest build, and that seems to work as well:

```powershell
PS /mnt/c/Windows/System32> install-module Microsoft.FastTrack.Migration -WhatIf
What if: Performing the operation "Install-Module" on target "Version '2.1.3535' of module 'Microsoft.FastTrack.Migration'".
What if: Performing the operation "Install-Module" on target "Version '2.0.2.4' of module 'AzureAD'".
PS /mnt/c/Windows/System32> install-module Microsoft.FastTrack.Migration -WhatIf  -AllowPrerelease
What if: Performing the operation "Install-Module" on target "Version '2.1.3536-alpha' of module 'Microsoft.FastTrack.Migration'".
What if: Performing the operation "Install-Module" on target "Version '1.2.0-preview' of module 'PackageManagement'".
What if: Performing the operation "Install-Module" on target "Version '2.0.2.4' of module 'AzureAD'".
```

So if there's a clear repro which still works let me know!

**Comment by sdaaish** [comment]
# Table of Contents

1.  [windows](#org7db5e58)
2.  [WSL](#orgbce8534)

I have the same problem installing [posh-git](https://github.com/dahlbyk/posh-git#installation) version 1 on WSL, Windows Subshell for Linux, on a host where I have it installed for windows.
Ie it's on the same computer.
I have version 1 installed at home on a pure linux-host, using `-AllowPreRelease`, and can't recall any problems with that.

It seems that I cannot see that there are other versions in the PSgallery than 0.7.3 with `find-module`.


<a id="org7db5e58"></a>

# windows

This is what I see from windows on the same host:

    C:\Users\sdaa
    04-05 11:47:39 13 > $psversiontable
    
    Name                           Value
    ----                           -----
    PSVersion                      6.2.0
    PSEdition                      Core
    GitCommitId                    6.2.0
    OS                             Microsoft Windows 10.0.17134
    Platform                       Win32NT
    PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
    PSRemotingProtocolVersion      2.3
    SerializationVersion           1.1.0.1
    WSManStackVersion              3.0
    
    C:\Users\sdaa
    04-05 11:47:29 11 > find-module -name posh-git -AllVersions -AllowPrerelease
    
    Version              Name                                Repository           Description
    -------              ----                                ----------           -----------
    1.0.0-beta3          posh-git                            PSGallery            Provides prompt with Git status summary information and tab completion for Git commands, parameters, remotes and branch names.
    1.0.0-beta2          posh-git                            PSGallery            Provides prompt with Git status summary information and tab completion for Git commands, parameters, remotes and branch names.
    1.0.0-beta1          posh-git                            PSGallery            Provides prompt with Git status summary information and tab completion for Git commands, parameters, remotes and branch names.
    0.7.3                posh-git                            PSGallery            Provides prompt with Git status summary information and tab completion for Git commands, parameters, remotes and branch names.
    0.7.2-pre1           posh-git                            PSGallery            Provides prompt with Git status summary information and tab completion for Git commands, parameters, remotes and branch names.
    0.7.1                posh-git                            PSGallery            Provides prompt with Git status summary information and tab completion for Git commands, parameters, remotes and branch names.
    0.7.0                posh-git                            PSGallery            Provides prompt with Git status summary information and tab completion for Git commands, parameters, remotes and branch names.
    0.6.1.20160330       posh-git                            PSGallery            A PowerShell environment for Git
    0.6.0.20160310       posh-git                            PSGallery            A PowerShell environment for Git
    0.5.0.2015           posh-git                            PSGallery            A set of PowerShell scripts which provide Git/PowerShell integration
    
    
    C:\Users\sdaa
    04-05 11:47:32 12 > Get-PSRepository
    
    Name                      InstallationPolicy   SourceLocation
    ----                      ------------------   --------------
    PSGallery                 Trusted              https://www.powershellgallery.com/api/v2/


<a id="orgbce8534"></a>

# WSL

And this is the same commands from `WSL` on the same host, running `Ubuntu1804`.

    PS /home/sdaa/tmp> $psversiontable
    
    Name                           Value
    ----                           -----
    PSVersion                      6.2.0
    PSEdition                      Core
    GitCommitId                    6.2.0
    OS                             Linux 4.4.0-17134-Microsoft #523-Microsoft Mon Dec 31 17:49:00 PST 2018
    Platform                       Unix
    PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
    PSRemotingProtocolVersion      2.3
    SerializationVersion           1.1.0.1
    WSManStackVersion              3.0
    
    PS /home/sdaa/tmp>
    
    PS /home/sdaa/tmp> find-module -name posh-git -AllVersions -AllowPrerelease
    
    Version              Name                                Repository           Description
    -------              ----                                ----------           -----------
    0.7.3                posh-git                            PSGallery            Provides prompt with Git status summary …
    
    PS /home/sdaa/tmp> Get-PSRepository
    
    Name                      InstallationPolicy   SourceLocation
    ----                      ------------------   --------------
    PSGallery                 Trusted              https://www.powershellgallery.com/api/v2
    
    PS /home/sdaa/tmp>

There is a difference in the URL for `PSGallery` where it has a trailing slash on windows and none on WSL. But I have tested to Unregister and Register the repository with the default settings with no chnage in the result.

**Comment by edyoung** [comment]
I believe this is the same root cause as #399 and should be resolved by https://github.com/OneGet/oneget/pull/451

**Comment by edyoung** [comment]
This is resolved by https://github.com/OneGet/oneget/pull/451

**Comment by edyoung** [comment]
fixed in 2.1.4/packagemanagement 1.4
